### PR TITLE
Deploy etcd CA cert if k3s use embeded etcd

### PR DESCRIPTION
**Pull request by mogliang**
This is part of etcd changes mentioned in https://github.com/cluster-api-provider-k3s/cluster-api-k3s/issues/75

To be able to access etcd, mgmt cluster need manage etcd server ca

### CODE REVIEW

File: pkg/secret/certificates.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ var (
 type Certificates []*Certificate
 
 // NewCertificatesForInitialControlPlane returns a list of certificates configured for a control plane node.
-func NewCertificatesForInitialControlPlane() Certificates {
+func NewCertificatesForInitialControlPlane(config *bootstrapv1.KThreesConfigSpec) Certificates {
```
Comment (zawachte) [inline pkg/secret/certificates.go:68]: is there something we can do for "join control planes" to allow existing clusters to use this feature?

Comment (mogliang) [reply pkg/secret/certificates.go:68]: The existing clusters etcd cert is stored in target cluster vm. There is no good way to export that cert back to mgmt cluster, unless user ssh vm, export file and create secret manually. Besides, k3s ca cert https://docs.k3s.io/cli/certificate is a bit different than what we create in k3s cp provider, adding possibility of human error and k3s versioning, which makes things more complex...

I would prefer adding flags in kcp cr, let it tell if it enabled etcd connection. New created cluster can enable it by set flag.

Comment (mogliang) [reply pkg/secret/certificates.go:68]: renaming KThreesConfigSpec helper method to `IsEtcdEmbedded`. Differentiate with controlplane method `IsEtcdManaged`

IsEtcdManaged shall be IsEtcdEmbedded && etcdCASecret exists.

**Review by zawachte — APPROVED** [review]

**Comment by zawachte** [comment]
The change looks good to me. The one caveat about this change is that existing clusters will need to live with no etcd certs located in the management cluster. This should be ok as long as the future PRs exist gracefully if the certificate secret doesn't exist.

**Pull request merged**